src/components/Chat/InputMessageV2.tsx

```diff
@@ -244,34 +244,39 @@ const InputMessageV2 = ({
         maxFiles: 20,
         sortOrder: 'asc',
         smartAlbums: ['RecentlyAdded', 'UserLibrary']
-      }).then(async (medias) => {
-        const newMedias: any = [];
-        for (const media of medias) {
-          if (media?.mime?.includes('video')) {
-            const thumbnailPath = await getVideoThumbnail(media.path);
-            newMedias.push({
-              type: 'video',
-              path: thumbnailPath,
-              pathVideo: media.path,
-              nameVideo: media.filename,
-              typeVideo: media.mime
-            });
-          } else {
-            const imageCropped = await ImagePicker.openCropper({
-              mediaType: 'photo',
-              path: media.path,
-              width: media.width,
-              height: media.height,
-              cropperChooseText: 'Next',
-              freeStyleCropEnabled: true
-            });
-            newMedias.push({type: 'image', path: imageCropped.path});
+      })
+        .then(async (medias) => {
+          setIsLoadingUploadImageMedia(false);
+          const newMedias: any = [];
+          for (const media of medias) {
+            if (media?.mime?.includes('video')) {
+              const thumbnailPath = await getVideoThumbnail(media.path);
+              newMedias.push({
+                type: 'video',
+                path: thumbnailPath,
+                pathVideo: media.path,
+                nameVideo: media.filename,
+                typeVideo: media.mime
+              });
+            } else {
+              const imageCropped = await ImagePicker.openCropper({
+                mediaType: 'photo',
+                path: media.path,
+                width: media.width,
+                height: media.height,
+                cropperChooseText: 'Next',
+                freeStyleCropEnabled: true
+              });
+              newMedias.push({type: 'image', path: imageCropped.path});
+            }
           }
-        }
 
-        refAttachment.current.close();
-        handleUploadMedia(newMedias);
-      });
+          refAttachment.current.close();
+          handleUploadMedia(newMedias);
+        })
+        .catch(() => {
+          setIsLoadingUploadImageMedia(false);
+        });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new code handles the loading state and error handling for media uploads. However, it seems to be missing a crucial part of the old code: permission checking. Before accessing the device's gallery, camera, or file picker, it's important to check if the app has the necessary permissions. If not, the user should be alerted and guided to adjust their permission settings. Here is how you can add it back:

```diff
+    const {success} = await requestExternalStoragePermission();
+    if (success) {
         ImagePicker.openPicker({
           maxFiles: 20,
           sortOrder: 'asc',
           smartAlbums: ['RecentlyAdded', 'UserLibrary']
         })
         .then(async (medias) => {
           setIsLoadingUploadImageMedia(false);
           // rest of the code...
         })
         .catch(() => {
           setIsLoadingUploadImageMedia(false);
         });
+    } else {
+      setIsLoadingUploadImageMedia(false);
+      openAlertPermission(
+        'We’re not able to access your photo library, please adjust your permission settings for BetterSocial.'
+      );
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->